common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -337,28 +351,32 @@ private boolean processInvalidUrl(@NonNull final WebView view, @NonNull final St
             return true;
         }
 
-        if (url.toLowerCase(Locale.US).equals("about:blank")) {
+        if ("about:blank".equals(lowerCaseUrl)) {
             Logger.verbose(TAG, "It is an blank page request");
             return true;
         }
 
-        if (!url.toLowerCase(Locale.US).startsWith(AuthenticationConstants.Broker.REDIRECT_SSL_PREFIX)) {
-            String redactedUrl = "redacted";
-            try {
-                redactedUrl = StringExtensions.removeQueryParameterFromUrl(url);
-            } catch (final URISyntaxException e) {
-                // Best effort.
-            }
+        if (!lowerCaseUrl.startsWith(AuthenticationConstants.Broker.REDIRECT_SSL_PREFIX)) {
+            String redactedUrl = removeQueryParameters(url);
 
             Logger.error(TAG, "The webView was redirected to an unsafe URL: " + redactedUrl, null);
             returnError(ErrorStrings.WEBVIEW_REDIRECTURL_NOT_SSL_PROTECTED, "The webView was redirected to an unsafe URL.");
             view.stopLoading();
             return true;
         }
-
+        Logger.infoPII(TAG,"We are declining to override loading and redirect to invalid URL: '"
+                + removeQueryParameters(url) + "' the user's url pattern is '" + mRedirectUrl + "'");
         return false;
     }
 
+    private String removeQueryParameters(@NonNull String url) {
```

Comment:
The method name can be a bit confusing since if we hit the `catch` block then we are not returning a URL with query parameters removed instead we are returning a pre-defined String

Reply:
removeQueryParametersOrRedact?

Reply:
Yup, that sounds good!

Reply:
changed.